lib/ts/recipe/accountlinking/recipeImplementation.ts

```diff
@@ -0,0 +1,634 @@
+            }
+            let accountsLinkingResult = await querier.sendPostRequest(
+                new NormalisedURLPath("/recipe/accountlinking/user/link"),
+                {
+                    recipeUserId,
+                    primaryUserId,
+                }
+            );
+            if (accountsLinkingResult.status === "OK") {
+                await Session.revokeAllSessionsForUser(recipeUserId, userContext);
+                let user: User | undefined = await this.getUser({
+                    userId: primaryUserId,
+                    userContext,
+                });
+                if (user === undefined) {
+                    throw Error("this error should never be thrown");
+                }
+                let loginMethodInfo = user.loginMethods.find((u) => u.recipeUserId === recipeUserId);
+                if (loginMethodInfo === undefined) {
+                    throw Error("this error should never be thrown");
+                }
+                let recipeUser = await SuperTokens.getInstanceOrThrowError()._getUserForRecipeId(
+                    loginMethodInfo.recipeUserId,
+                    loginMethodInfo.recipeId
+                );
+                if (recipeUser.user === undefined) {
+                    throw Error("this error should never be thrown");
+                }
+                await config.onAccountLinked(user, recipeUser.user, userContext);
+            } else {
+                throw Error(`error thrown from core while linking accounts: ${accountsLinkingResult.status}`);
+            }
+            return accountsLinkingResult;
+        },
+        unlinkAccounts: async function (
+            this: RecipeInterface,
+            {
+                recipeUserId,
+                userContext,
+            }: {
+                recipeUserId: string;
+                userContext: any;
+            }
+        ): Promise<{
+            status: "OK";
+            wasRecipeUserDeleted: boolean;
+        }> {
+            let recipeUserIdToPrimaryUserIdMapping = await this.getPrimaryUserIdsforRecipeUserIds({
+                recipeUserIds: [recipeUserId],
+                userContext,
+            });
+            let primaryUserId = recipeUserIdToPrimaryUserIdMapping[recipeUserId];
+            if (primaryUserId === undefined || primaryUserId === null) {
+                throw Error("recipeUserId is not associated with any primaryUserId");
+            }
+            /**
+             * primaryUserId === recipeUserId
+             */
+            if (primaryUserId === recipeUserId) {
+                let user = await this.getUser({
+                    userId: primaryUserId,
+                    userContext,
+                });
+
+                if (user === undefined) {
+                    return {
+                        status: "OK",
+                        wasRecipeUserDeleted: false,
+                    };
+                }
+                if (user.loginMethods.length > 1) {
```

Comment:
i don't think the logic for if this is false is correct - the lucid chart diagram also seems wrong here